cranelift/codegen/src/ir/stackslot.rs

```diff
@@ -69,6 +69,37 @@ pub struct StackSlotData {
     /// be aligned according to other considerations, such as minimum
     /// stack slot size or machine word size, as well.
     pub align_shift: u8,
+
+    /// Opaque stackslot metadata handle, passed through to
+    /// compilation result metadata describing stackslot location.
+    ///
+    /// In the face of compiler transforms like inlining that may move
+    /// stackslots between functions, when an embedder wants to
+    /// externally observe stackslots, it needs a first-class way for
+    /// the identity of stackslots to be carried along with the IR
+    /// entities. This opaque `StackSlotKey` allows the embedder to do
+    /// so.
+    pub key: Option<StackSlotKey>,
```

Comment:
Maybe use PackedOption?

Reply:
I don't necessarily want to take any bites out of the domain of the arbitrary u64 -- it's already used by Wasmtime to carry a FuncKey which has its own bit-level encoding (rather than a serial index). Hopefully though stackslots are sparse enough (vs. e.g. instructions or blocks) that this is OK...